lib/services/auth/auth_service.dart

```diff
@@ -0,0 +1,98 @@
+import 'dart:convert';
+
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:survey/services/api/api_service.dart';
+import 'package:survey/services/api/auth/auth_api_service.dart';
+import 'package:survey/services/api/user/user_api_service.dart';
+import 'package:survey/services/locator/locator_service.dart';
+import 'package:survey/models/auth_token_info.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/models/user_info.dart';
+
+part 'auth_local_storage.dart';
+
+abstract class AuthService {
+  bool get isAuthenticated;
+
+  String? get accessToken;
+
+  UserInfo? get user;
+
+  Future<void> login({
+    required AuthLoginParams params,
```

Comment:
I think the `AuthLoginParams` shouldn't be here, it's being introduced at `AuthApiService`. This AuthService is called from your application layer, which contains the UI, which is usually passing in the String forms of data (e.g: email, password....etc...). So I'd recommend passing them in here as String, and then inside here, you can wrap them into `AuthLoginParams` and pass them to `AuthApiService`. Why? because later if you would like to change the implementation of `AuthApiService` for example adding a new field that is not visible from the UI, then you don't have to change the app layer code, instead, you will have to only update at this point, where we are changing from type X -> type Y (separation of concern).

Reply:
@sleepylee  I updated it 👍